lib/stat_tracker.rb

```diff
@@ -0,0 +1,31 @@
+require 'csv'
+
+class StatTracker
+
+  def self.from_csv(files)
+    data = {}
+    files.each { |key, val| data[key] = CSV.read(val) }
+    data
```

Comment:
what's the end result of the ::from_csv method? isn't it supposed to create objects for each team, game or game teams? so we have to iterate over the locations hash, creating a bunch of objects for each iteration and saving them so we can then continue to use the data right? so i think we need to use map instead of each?

Reply:
I don't know how well map works when storing the block code in a hash. The output of this method is a hash with 3 keys (the nicknames of the three data files) and the values are the csv data, which are stored as nested arrays, each element is one line from the csv file. The problem with is that this method converts absolutely everything to strings, whereas the table preserves the numeric values in the #from_csv_2 method.